src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -140,6 +142,30 @@ Survey getOne() {
             }
             return surveyMapper.queryPage(DynamoSurvey.class, query).getResults();
         }
+        
+        private List<DynamoSurvey> queryBySurveyIdentifier() {
+            if (studyIdentifier == null) {
+                throw new IllegalStateException("Calculated the need to query by secondary index, but study identifier is not set");
+            }
+            DynamoSurvey hashKey = new DynamoSurvey();
+            hashKey.setStudyIdentifier(studyIdentifier);
+            hashKey.setIdentifier(surveyGuid.substring(11));
+            DynamoDBQueryExpression<DynamoSurvey> query = new DynamoDBQueryExpression<DynamoSurvey>();
+            query.withHashKeyValues(hashKey);
+            query.withConsistentRead(false);
+            query.withRangeKeyCondition("identifier", equalsString(surveyGuid.substring(11)));
+            if (published) {
+                query.withQueryFilterEntry(PUBLISHED_PROPERTY, equalsNumber("1"));
+            }
+            if (notDeleted) {
+                query.withQueryFilterEntry(DELETED_PROPERTY, equalsNumber("0"));
+            }
+            if (createdOn != 0L) {
+                query.withRangeKeyCondition(CREATED_ON_PROPERTY, equalsNumber(Long.toString(createdOn)));
+            }
+            
+            return surveyMapper.queryPage(DynamoSurvey.class, query).getResults();
```

Comment:
Querying by guid returns the surveys from newest to oldest (descending createdOn order). Since we don't have the range key to do that for us, we'll need to sort the results ourselves before returning them.